app/data_model/answer_store.py

```diff
@@ -42,97 +65,108 @@ def _validate(answer):
-            escaped.append(answer)
+            answer = copy(answer)
+            if isinstance(answer.value, str):
+                answer.value = escape(answer.value)
+            escaped.append(answer.to_dict())
         return self.__class__(existing_answers=escaped)
 
-    def filter(self, answer_ids):
+    def get_answer(self, answer_id: str, list_item_id: str = None) -> Optional[Answer]:
+        """ Get a single answer from the store
+
+        Args:
+            answer_id: The answer id to find
+            list_item_id: If not provided (None), will only match an answer with no list_item_id
+
+        Returns:
+            A single Answer or None if it doesn't exist
         """
-        Find all answers in the answer store for a given set of filter parameter matches.
+        answer = self.answer_map.get((answer_id, list_item_id))
 
-        :param answer_ids: The answer ids to filter results by
-        :return: Return a new AnswerStore object with filtered answers for chaining
+        if answer:
+            return Answer.from_dict(answer)
+
+    def get_answers_by_answer_id(self, answer_ids: List[str], list_item_id: str = None) -> List[Answer]:
+        """ Get multiple answers from the store using the answer_id
+
+        Args:
+            answer_ids: list of answer ids to find
+            list_item_id: list item id to match
+                          If not provided (None), will only match an answer with no list_item_id
+
+        Returns:
+            A list of Answer objects
         """
-        answers = [v for k, v in self.answer_map.items() if k in answer_ids]
+        output = []
+        for answer_id in answer_ids:
+            answer = self.answer_map.get((answer_id, list_item_id))
+            if answer:
+                output.append(answer)
 
-        return self.__class__(existing_answers=answers)
+        return AnswerStore(output)
 
     def clear(self):
         """
         Clears answers *in place*
         """
         self.answer_map.clear()
 
-    def remove(self, answer_ids=None):
+    def remove_answer(self, answer_id: str, list_item_id: str = None):
         """
-        Removes answer(s) *in place* from the answer store.
-
-        :param answer_ids: The answer ids to remove
+        Removes answer *in place* from the answer store.
         """
-        for answer_id in answer_ids:
-            if answer_id in self.answer_map:
-                del self.answer_map[answer_id]
 
-    def remove_answer(self, answer):
-        """
-        Removes answer *in place* from the answer store.
+        if self.answer_map.get((answer_id, list_item_id)):
+            del self.answer_map[(answer_id, list_item_id)]
+
+    def remove_all_answers_for_list_item_id(self, list_item_id: str):
+        """Remove all answers associated with a particular list_item_id
+        This method iterates through the entire list of answers.
 
-        :param answer: The answer to remove
+        *Not efficient.*
```

Comment:
Would it be more efficient to iterate the list first to identify the items to delete, then loop over those and delete in place?

Reply:
Sorted I think